app/notification.go

```diff
@@ -15,13 +15,138 @@ import (
 	"github.com/mattermost/mattermost-server/store"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/mattermost/mattermost-server/utils/markdown"
+	"github.com/mitchellh/mapstructure"
 )
 
 const (
 	THREAD_ANY  = "any"
 	THREAD_ROOT = "root"
 )
 
+// TODO : Comment
+func (a *App) SendEphemeralPostWrapper(userId string, postDetails interface{}, post model.Post) {
+	err := mapstructure.Decode(postDetails, &post)
+	if err != nil {
+		panic(nil)
+	}
+	a.SendEphemeralPost(userId, &post)
+}
+
+// AddMentionedUsers will add the mentioned user id in the struct's list for mentioned users
+func (e *ExplicitMentions) AddMentionedUsers(ids []string) {
+	for _, id := range ids {
+		(*e).MentionedUserIds[id] = true
```

Comment:
All this de-referencing to access properties is not necessary. See here: https://tour.golang.org/moretypes/4

Reply:
Yeah I already changed it for (*e) to just e. Shortly after the PR I realised that they do the same thing.